cluster-agent/controllers/argoproj.io/application_controller_test.go

```diff
@@ -0,0 +1,539 @@
+	return reconcile.Request{
+		NamespacedName: types.NamespacedName{
+			Name:      name,
+			Namespace: namespace,
+		},
+	}
+}
+
+// Create sample data for testing.
+func createSampleData(dbq db.AllDatabaseQueries) (*db.ClusterCredentials, *db.ManagedEnvironment, *db.GitopsEngineCluster, *db.GitopsEngineInstance, *db.ClusterAccess, error) {
+
+	ctx := context.Background()
+	var err error
+
+	clusterCredentials, managedEnvironment, engineCluster, engineInstance, clusterAccess := generateSampleData()
+
+	if err = dbq.CreateClusterCredentials(ctx, &clusterCredentials); err != nil {
+		return nil, nil, nil, nil, nil, err
+	}
+
+	if err = dbq.CreateManagedEnvironment(ctx, &managedEnvironment); err != nil {
+		return nil, nil, nil, nil, nil, err
+	}
+
+	if err = dbq.CreateGitopsEngineCluster(ctx, &engineCluster); err != nil {
+		return nil, nil, nil, nil, nil, err
+	}
+
+	if err = dbq.CreateGitopsEngineInstance(ctx, &engineInstance); err != nil {
+		return nil, nil, nil, nil, nil, err
+	}
+
+	if err = dbq.CreateClusterAccess(ctx, &clusterAccess); err != nil {
+		return nil, nil, nil, nil, nil, err
+	}
+
+	return &clusterCredentials, &managedEnvironment, &engineCluster, &engineInstance, &clusterAccess, nil
+
+}
+
+var testClusterUser = &db.ClusterUser{
+	Clusteruser_id: "test-user",
+	User_name:      "test-user",
+}
+
+// Generate fake cluster for testing.
+func generateSampleData() (db.ClusterCredentials, db.ManagedEnvironment, db.GitopsEngineCluster, db.GitopsEngineInstance, db.ClusterAccess) {
+	clusterCredentials := db.ClusterCredentials{
+		Clustercredentials_cred_id:  "test-cluster-creds-test",
+		Host:                        "host",
+		Kube_config:                 "kube-config",
+		Kube_config_context:         "kube-config-context",
+		Serviceaccount_bearer_token: "serviceaccount_bearer_token",
+		Serviceaccount_ns:           "Serviceaccount_ns",
+	}
+
+	managedEnvironment := db.ManagedEnvironment{
+		Managedenvironment_id: "test-managed-env",
+		Clustercredentials_id: clusterCredentials.Clustercredentials_cred_id,
+		Name:                  "my env",
+	}
+
+	gitopsEngineCluster := db.GitopsEngineCluster{
+		Gitopsenginecluster_id: "test-fake-cluster",
+		Clustercredentials_id:  clusterCredentials.Clustercredentials_cred_id,
+	}
+
+	gitopsEngineInstance := db.GitopsEngineInstance{
+		Gitopsengineinstance_id: "test-fake-engine-instance-id",
+		Namespace_name:          "test-fake-namespace",
+		Namespace_uid:           "test-fake-namespace",
+		EngineCluster_id:        gitopsEngineCluster.Gitopsenginecluster_id,
+	}
+
+	clusterAccess := db.ClusterAccess{
+		Clusteraccess_user_id:                   testClusterUser.Clusteruser_id,
+		Clusteraccess_managed_environment_id:    managedEnvironment.Managedenvironment_id,
+		Clusteraccess_gitops_engine_instance_id: gitopsEngineInstance.Gitopsengineinstance_id,
+	}
+
+	return clusterCredentials, managedEnvironment, gitopsEngineCluster, gitopsEngineInstance, clusterAccess
+}
+
+func testSetup() error {
```

Comment:
I hope you would have confirmed through ./psql.sh or pgadmin, that all tables are being cleaned up.

Reply:
Yes I did @jparsai